earth_enterprise/src/fusion/autoingest/AssetSerializer.h

```diff
@@ -0,0 +1,55 @@
+#ifndef ASSET_SERIALIZER_H
+#define ASSET_SERIALIZER_H
+
+#include "common/khxml/khxml.h"
+#include "common/khxml/khdom.h"
+
+using namespace khxml;
+
+template<class AssetType>
+class AssetSerializerLocalXML
+{
+  public:
+    AssetSerializerLocalXML() {}
+    AssetPointerType<AssetType> Load(const std::string &boundref)
+    {
+      return AssetType::Load(boundref);
+    }
+
+    bool Save(AssetPointerType<AssetType> asset, std::string filename){
+      using AssetStorageType = typename AssetType::Base;
+      extern void ToElement(DOMElement *elem, const AssetStorageType &self);
+
+      std::unique_ptr<GEDocument> doc = CreateEmptyDocument(asset->GetName());
+      if (!doc) {
+          notify(NFY_WARN,
+                "Unable to create empty document: %s", asset->GetName().c_str());
+          return false;
+      }
+      bool status = false;
+      try {
+          DOMElement *top = doc->getDocumentElement();
+          if (top) {
+              const AssetStorageType &storage = *asset;
+              ToElement(top, storage);
+              asset->SerializeConfig(top);
+              status = WriteDocument(doc.get(), filename);
+
+              if (!status && khExists(filename)) {
+                  khUnlink(filename);
+              }
+          } else {
+              notify(NFY_WARN, "Unable to create document element %s",
+                    filename.c_str());
+          }
+      } catch (const std::exception &e) {
```

Comment:
I think it would be a good idea to keep the full error handling here (i.e., catch XMLException and DOMException separately).